server/services/untranslated.js

```diff
@@ -0,0 +1,124 @@
+  return {
+    /**
+     * Get an untranslated source entity based not the parameters
+     *
+     * @param {object} param0 Parameters for the content type
+     * @param {object} param1 Parameters for the query result
+     * @returns One entity of the content type 'uid' that has not yet been translated from 'sourceLocale' to 'targetLocale'
+     */
+    async getUntranslatedEntity(
+      { uid, targetLocale, sourceLocale },
+      { populate }
+    ) {
+      const {
+        collectionName,
+        info: { singularName },
+      } = strapi.contentTypes[uid]
+      const notTranslated = await strapi.db
+        .getConnection(collectionName)
+        .select(`${collectionName}.id`)
+        // Join the other localizations (null allowed)
+        .leftJoin(
+          `${collectionName}_localizations_links as l`,
+          `${collectionName}.id`,
+          `l.${singularName}_id`
+        )
+        .leftJoin(
+          `${collectionName} as c2`,
+          `l.inv_${singularName}_id`,
+          'c2.id'
+        )
+        // The other localizations should not include the target locale
+        .whereNotIn(
+          `${collectionName}.id`,
+          strapi.db
+            .getConnection(collectionName)
+            .select(`${collectionName}.id`)
+            // Join the other localizations (null not allowed)
+            .join(
+              `${collectionName}_localizations_links as l`,
+              `${collectionName}.id`,
+              `l.${singularName}_id`
+            )
+            .join(
+              `${collectionName} as c2`,
+              `l.inv_${singularName}_id`,
+              'c2.id'
+            )
+            // other localization should be the target
+            .where('c2.locale', targetLocale)
+            // start localization should be the source
+            .andWhere(`${collectionName}.locale`, sourceLocale)
+        )
+        // Only from the source locale
+        .andWhere(`${collectionName}.locale`, sourceLocale)
+        // Only the first needed
+        .limit(1)
+      if (!notTranslated || notTranslated.length == 0) {
+        return null
+      }
+      // Fetch the whole data
+      return strapi.db.query(uid).findOne({
+        where: { id: notTranslated[0].id },
+        populate,
+      })
+    },
+
+    /**
+     * Calculate if a locale is fully translated,
+     * i.e. there are no other entities in any other locale
+     * that do not have a localization in this locale
+     *
+     * @param {string} uid Content-Type-UID
+     * @param {string} targetLocale the target locale
+     * @returns if the target locale is fully translated
+     */
+    async ifFullyTranslated(uid, targetLocale) {
+      const {
+        collectionName,
+        info: { singularName },
+      } = strapi.contentTypes[uid]
+      const notTranslated = await strapi.db
+        .getConnection(collectionName)
+        // Join the other localizations (null allowed)
+        .leftJoin(
```

Comment:
Fails somewhere around here for contentTypes comprised of > 1 word, I think this is because collectionName has "_" as a seperator while singularName uses "-".

<details><summary>Logs</summary>
<p>
```
[2022-05-17 21:02:54.749] error: select `hello_worlds`.`id` from `hello_worlds` left join `hello_worlds_localizations_links` as `l` on `hello_worlds`.`id` = `l`.`hello-world_id` left join `hello_worlds` as `c2` on `l`.`inv_hello-world_id` = `c2`.`id` - no such column: l.hello-world_id
SqliteError: select `hello_worlds`.`id` from `hello_worlds` left join `hello_worlds_localizations_links` as `l` on `hello_worlds`.`id` = `l`.`hello-world_id` left join `hello_worlds` as `c2` on `l`.`inv_hello-world_id` = `c2`.`id` - no such column: l.hello-world_id
    at Database.prepare (/home/layaxx/Documents/programming/strapi/my-project/node_modules/better-sqlite3/lib/methods/wrappers.js:5:21)
    at Client_BetterSQLite3._query (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/dialects/better-sqlite3/index.js:30:34)
    at executeQuery (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
    at Client_BetterSQLite3.query (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/client.js:146:12)
    at Runner.query (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/execution/runner.js:130:36)
    at ensureConnectionCallback (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/execution/internal/ensure-connection-callback.js:13:17)
    at Runner.ensureConnection (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/execution/runner.js:307:20)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Runner.run (/home/layaxx/Documents/programming/strapi/my-project/node_modules/knex/lib/execution/runner.js:30:19)
    at async Object.ifFullyTranslated (/home/layaxx/Documents/programming/strapi/my-project/node_modules/strapi-plugin-deepl/server/services/untranslated.js:88:29)
    at async /home/layaxx/Documents/programming/strapi/my-project/node_modules/strapi-plugin-deepl/server/services/translate.js:72:30
    at async Promise.all (index 0)
    at async /home/layaxx/Documents/programming/strapi/my-project/node_modules/strapi-plugin-deepl/server/services/translate.js:67:22
    at async Promise.all (index 1)
    at async Object.contentTypes (/home/layaxx/Documents/programming/strapi/my-project/node_modules/strapi-plugin-deepl/server/services/translate.js:56:21)
    at async Object.batchTranslateContentTypes (/home/layaxx/Documents/programming/strapi/my-project/node_modules/strapi-plugin-deepl/server/controllers/translate.js:205:13)```

</p>
</details>

Reply:
_Fails_:
```select `hello_worlds`.`id` from `hello_worlds` left join `hello_worlds_localizations_links` as `l` on `hello_worlds`.`id` = `l`.`hello-world_id` left join `hello_worlds` as `c2` on `l`.`inv_hello-world_id` = `c2`.`id`  ```
_Succeeds_:
```select `hello_worlds`.`id` from `hello_worlds` left join `hello_worlds_localizations_links` as `l` on `hello_worlds`.`id` = `l`.`hello_world_id` left join `hello_worlds` as `c2` on `l`.`inv_hello_world_id` = `c2`.`id` ```

Not sure how best to highlight, but the second version has `hello_world_id` instead of `hello-world_id` twice